### PR TITLE
doc: add reporting bug doc

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/911

/cc @robszumski @kelseyhightower @brianredbeard

### CODE REVIEW

File: Documentation/reporting_bug.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+### How to get etcd configuration and log when it runs as systemd service ‘etcd2.service’
+
+``` bash
+$ sudo systemctl cat etcd2
+$ sudo journalctl -u etcd2
+```
+
+Due to some systemd reason, journald may miss the last few log lines when its process exit. If journalctl tells you that etcd stops without fatal or panic message, you could try `sudo journalctl -f -t etcd2` to get full log.
```
Comment (robszumski) [inline Documentation/reporting_bug.md]: Due to an upstream systemd bug...

File: README.md [hunk 2]
```diff
@@ -99,6 +99,10 @@ Now it's time to dig into the full etcd API and other guides.
 
 See [CONTRIBUTING](CONTRIBUTING.md) for details on submitting patches and the contribution workflow.
 
+## Reporting bugs
+
+See [reporting bug](Documentation/reporting_bug.md) for details on submitting bug reports workflow.
```
Comment (robszumski) [inline README.md]: ...for details about reporting any issue you may encounter.

File: Documentation/reporting_bug.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+## Reporting Bug
```
Comment (robszumski) [inline Documentation/reporting_bug.md]: Bugs (plural)

Comment (robszumski) [inline Documentation/reporting_bug.md]: or "Reporting a Bug"

Comment (robszumski) [inline Documentation/reporting_bug.md]: Might be good to rename this doc to match the title

**Comment by xiang90** [comment]
@robszumski All fixed. PTAL.

File: Documentation/reporting_bugs.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+## Reporting Bugs
+
+If you find bugs, documentation mistakes in etcd project, please let us know by [opening an issue](https://github.com/coreos/etcd/issues/new). We treat bugs and mistakes very seriously and believe no issue is too small. Before creating a bug report, please check there that one does not already exist.
```
Comment (robszumski) [inline Documentation/reporting_bugs.md]: ..bugs **or** documentation...

File: Documentation/reporting_bugs.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+To make your bug report accurate and easy to understand, please try to create bug reports that are:
+
+- Specific. Include as much details as possible: which version, what environment, what configuration, etc. You can also attach etcd log (the starting log with etcd configuration is especially important).
+
+- Reproducible. Include the steps to reproduce the problem. We understand some issues might be hard to reproduce, please includes the steps that might lead to the problem. You can also attach the affected etcd data dir and stack strace to the bug report.
+
+- Isolated. Please try to isolate and reproduce the bug with minimum dependencies. It would significantly slow down the speed to fix a bug if too many dependencies are involved in a bug report. We would rather not to debug an external system that uses etcd.
```
Comment (robszumski) [inline Documentation/reporting_bugs.md]: let's be a little nicer about the external system thing:

"Debugging external systems that rely on etcd is out of scope, but we are happy to point you in the right direction or help you interact with etcd in the correct manner."

Comment (xiang90) [inline Documentation/reporting_bugs.md]: agree

**Comment by yichengq** [comment]
we closed this PR because we use master branch as the base branch for development instead of dev-2.2 after etcd 2.1 is cut. we will cherry-pick these commits and send new PR compared to master branch. Sorry for the inconvenience.

**Pull request closed without merging**